### PR TITLE
[ffmpeg] Update to 5.0.2.

### LINKED ISSUE #28706: [ffmpeg] videotoolbox unavailable on macOS
**Describe the bug**
videotoolbox is no longer available since ffmpeg upgrade to 5.0

**Environment**
- OS: macOS

### PR DESCRIPTION
**Pull request by Sibras**
Updates ffmpeg to 5.0.2.

Adds OS level detection for OSX (fixes #28706)

### CODE REVIEW

**Comment by xiaozhuai** [comment]
@Sibras Would you please check this issue #28706? Thanks!

File: ports/ffmpeg/portfile.cmake [hunk 1]
```diff
@@ -17,17 +17,15 @@ vcpkg_from_github(
...
     message(FATAL_ERROR "Error: ffmpeg will not build with spaces in the path. Please use a directory with no spaces")
 endif()
 
-
 if(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
-    # ffmpeg nasm build gives link error on x86, so fall back to yasm
+    # ffmpeg nasm build causes linker to stall on x86, so fall back to yasm
     vcpkg_find_acquire_program(YASM)
     get_filename_component(YASM_EXE_PATH "${YASM}" DIRECTORY)
     vcpkg_add_to_path("${YASM_EXE_PATH}")
```
Comment (Neumann-A) [inline ports/ffmpeg/portfile.cmake:31]: OT: can this be replaced with clang in the future? (e.g. see `gmp`)

Comment (Sibras) [reply ports/ffmpeg/portfile.cmake:31]: clang currently doesnt support netwide syntax which is used in ffmpeg asm source.

Comment (Neumann-A) [reply ports/ffmpeg/portfile.cmake:31]: Is it really netwide syntax or Intel syntax? After googling, the latter could probably be dealt with using `-x86-asm-symtab=intel` (or `--x86-asm-syntax=intel`)

Comment (Sibras) [reply ports/ffmpeg/portfile.cmake:31]: netwide is very similar to intel syntax for the actual instructions but it has additional functionality that requires a netwide compatible pre-processing pass

Comment (Neumann-A) [reply ports/ffmpeg/portfile.cmake:31]: yeah it fails already due to `;` being used for comments

**Review by FrankXie05 — APPROVED** [review]

**Comment by FrankXie05** [comment]
All features and new options have been tested successfully.

**Pull request merged**